src/transaction.js

```diff
@@ -120,8 +120,13 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
   }
 
   if (hasWitnesses) {
+    var isNull = true
     for (i = 0; i < vinLen; ++i) {
       tx.ins[i].witness = readVector()
+      isNull = isNull && tx.ins[i].witness.length === 0
+    }
+    if (isNull) {
+      throw new Error('Transaction has superfluous witness data')
```

Comment:
Do we need this?
I understand its not ideal,  but,  it doesn't really hurt any one.